### ISSUE TITLE
Pytorch-Wildlife: A Collaborative Deep Learning Framework for Conservation

### ISSUE THREAD

**Issue by narausch** [pytorch-fdn/ecosystem#7]
### Contact emails
[zhongqimiao@microsoft.com](mailto:zhongqimiao@microsoft.com)

### Project summary
an open-source deep learning platform designed for creating, modifying, and sharing powerful AI models

### Project description
The rapid decline in biodiversity highlights the urgent need for large-scale wildlife monitoring. Deep-learning is applied to speed up data processing. Though promising, technical challenges often hinder its wider application. We herein introduce PyTorch-Wildlife, an open-source deep learning platform designed for creating, modifying, and sharing powerful AI models. Building on MegaDetector, PyTorch-Wildlife prioritizes usability and accessibility, ensuring ease of use to non-engineers. Its modular codebase also promotes streamlined development of new features.
One of Pytorch-Wildlife's standout features is the ease with which models can be loaded from its model zoo, often requiring just a few lines of code. Pytorch-Wildlife also offers an intuitive, user-friendly interface, accessible either through local installation or Hugging Face, for animal detection and classification in images and videos. The Pytorch-Wildlife framework includes a fine-tuning classification module that allows scientists to train custom animal classification models.
Pytorch-Wildlife has been effectively utilized in real-world applications for animal classification in the Amazon Rainforest and for detecting invasive opossums in the Galapagos Islands. Specifically, the opossum detection model achieves a 98% accuracy, while the Amazon Rainforest model can recognize 36 different animal species with 92% accuracy across 90% of the data, making the data annotation process 10 times faster. These examples illustrate the platform's potential to expedite research and aid ecologists in focusing on deeper tasks such as population density and biodiversity estimation.
Pytorch-Wildlife is available for installation on all operating systems and continues to evolve as a comprehensive tool for the conservation community, available at https://github.com/microsoft/CameraTraps/

### Are there any other projects in the PyTorch Ecosystem similar to yours? If, yes, what are they?
Not answered

### Project repo URL
https://github.com/microsoft/CameraTraps

### Additional repos in scope of the application
No response

### Project license
MIT

### GitHub handles of the project maintainer(s)
zhmiao

### Is there a corporate or academic entity backing this project? If so, please provide the name and URL of the entity.
Microsoft

### Website URL
No response

### Documentation
[Guides (Tutorials): https://torch-geopooling.readthedocs.io/en/latest/guide/index.html

### API Reference: 
[https://torch-geopooling.readthedocs.io/en/latest/reference/nn.html](https://github.com/microsoft/CameraTraps/blob/main/INSTALLATION.md)](https://torch-geopooling.readthedocs.io/en/latest/reference/nn.html%5D(https://github.com/microsoft/CameraTraps/blob/main/INSTALLATION.md))

### How do you build and test the project today (continuous integration)? Please describe.
We have internal testing pipelines.

### Version of PyTorch
No response

### Components of PyTorch
Project does not support TorchScript at this moment.

### How long do you expect to maintain the project?
At least 5 years from now. It will be extended in the future.

### Additional information
No response

**Comment by lantiga** [comment]
We'll be happy to accept the project, there's one aspect that needs to be addressed.

The LICENSE for the project is MIT, but it has ultralytics-yolov5 as a dependency and that is AGPL 3.0, which is incompatible.

To fix this, you'd probably need to keep the yolov5 dependency as one that is installed by the user, or change your own license to be compatible.

**Comment by thesteve0** [comment]
@zhmiao @danielaruizl1 @agentmorris 
Hey Folks - is there someone who can address the issue brought up in the comment above. We would like to finalize your inclusion in the PyTorch Ecosystem but we need the issue addressed

**Comment by zhmiao** [comment]
Hello @thesteve0 , thanks for checking in! And sorry for the late reply. We will probably drop the dependence of yolov5 in the next release and ask users to install that as needed. Besides yolov5, is there any other dependencies that might be an issue?

**Comment by thesteve0** [comment]
Ultralytics is also aGPL licensed so that should be removed as well. 

You might want to pin chardet to one of the most recent versions if it doesn't cause a conflict. 

```
chardet (Potential Conflict - LGPL)

License: LGPL-2.1 (versions < 7.0) / 0BSD & MIT (versions >= 7.0)

Details: Earlier versions of the chardet library were licensed under the GNU Lesser General Public License (LGPL). While LGPL is generally compatible with MIT projects if dynamically linked, strict corporate compliance checks often flag it as a risk. Luckily, starting with version 7.0, chardet was completely rewritten and relicensed to highly permissive 0BSD/MIT. If you want to avoid copyleft entirely, you should pin this dependency to chardet>=7.0.0 or replace it.
```

**Comment by thesteve0** [comment]
I know UltraLytics is a harder remove for you but I don't think we can proceed until you do. @michellearoth is there some sort of provisional status we can give them?

**Comment by thesteve0** [comment]
@zhmiao @narausch We had clarification on the license issues and that no longer is an issue. 
Unfortunately, PyTorch is a trademarked word and will need to be changed regardless of you joining the ecosystem. We are currently investigating if Torch is also trademarked. Once the name change is finished we should be able to add you to the ecosystem.

**Comment by michellearoth** [comment]
@zhmiao Hi there, please let us know when you update the project to remove "PyTorch" to comply with our trademark requirements. Thanks!

**Comment by zhmiao** [comment]
@michellearoth , yes, we will be changing names. Thanks for the reminder!

**Comment by michellearoth** [comment]
Hi @zhmiao do you have an update on the name change?

**Comment by zhmiao** [comment]
Hello @michellearoth, yes, we have halted all development on the project currently and will likely change the name in the near future. Now we are just waiting for feedback from our internal legal team for the final planning. We will keep you posted on this. Thanks.
